content/docs/use-cases/ci-and-ml.md

```diff
@@ -0,0 +1,171 @@
+to a dedicated testing platform.
+
+![](/img/ci_for_ml_retrain_model.jpg) _Model training flow_
+
+Assuming that in a project’s repository a `dvc` pipeline is defined for training
+a model. You can ask your CI tool to call `dvc pull` and, voilà, it is now has
+access to the _right_ data needed for training. The reproducing pipeline can be
+arbitrary complicated and comprise of many steps; regardless of all, `dvc repro`
+will yield a newly trained model. This new model will be tracked and pushed to
+the remote cache (note `dvc push`) and will be "linked" to the PR.
+
+```yml
+# .github/workflows/train_model.yml
+
+name: Retrain model
+on:
+  pull_request:
+    types: synchronize
+
+jobs:
+  train:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up the environment on the runner
+        run: |
+          apt-get update -y
+          apt-get install python3-dev -y
+          pip install -r requirements.txt
+
+      - name: DVC fetch the data used by the branch
+        run: dvc pull
+
+      - name: DVC retrain the model
+        run: dvc repro
+
+      - name: Track the newly trained model
+        run: |
+          dvc add model.pkl
+          git config --global user.name 'Your Name'
+          git config --global user.email 'my@enail.com'
+          git add model.pkl.dvc
+          git commit -m "Model retrained upon sync to PR"
+          git push
+          dvc phsh
+```
+
+### Bonus
+
+It might be that the training needs a tailored environment. This can be achieved
+by defining a runner that accommodates the required environment (for example a
+GPU).
+
+## Seamless model evaluation
+
+![](/img/ci_for_ml_evaluate_models.jpg) _Evaluating different models against the
+same validation dataset_
+
+An integral part of a prediction model’s quality, except its underlying code
+that can and should be tested, is its performance. Just like you wouldn’t merge
+code that fails tests into your `master` branch, you would not want to adopt a
+model that under-performs in comparison to the one you already have deemed
+stable on the `master` branch. It is common that one or more developers are
+maintaining or trying to improve a model that’s considered stable. In this case,
+before replacing this model with a new one, you would like to evaluate the new
+model’s performance and compare it to the performance of the “stable” model. For
+this comparison to be meaningful, you would like to evaluate the performances
+against the _same_ validation set.
+
+```yml
+name: Evaluate model
+on:
+  pull_request:
+    types: synchronize
+
+jobs:
+  evaluate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up the environment on the runner
+        run: |
+          apt-get update -y
+          apt-get install python3-dev -y
+          pip install -r requirements.txt
+
+      - name: DVC fetch the data used by the branch
+        run: dvc pull
```

Comment:
Probably this line has to be refactored so it will fetch the right validation data set.

Reply:
@pared I think here I need a little help in and make it clearer that which version of the data is brought to the workspace.

Reply:
To make it more readable I would consider `pull`ing specified target. To reproduce, I presume only data is needed (depending on how the project is structured). So, what we could do is `dvc pull data_eval.dvc` and then, in repro `dvc repro {eval_stage_name}` to stress where is actually important here.